// vuepress-theme-hope/lib/node/plugins/blog.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.resolveBlogOptions = exports.getBlogOptions = void 0;
const defaultOptions = {
    article: "/article/",
    category: "/category/",
    categoryItem: "/category/:name/",
    tag: "/tag/",
    tagItem: "/tag/:name/",
    encrypted: "/encrypted/",
    slides: "/slides/",
    star: "/star/",
    timeline: "/timeline/",
};
const compareDate = (dateA, dateB) => {
    if (!dateA)
        return 1;
    if (!dateB)
        return -1;
    return dateB.getTime() - dateA.getTime();
};
const sorter = (pageA, pageB) => {
    const prevKey = pageA.frontmatter.sticky;
    const nextKey = pageB.frontmatter.sticky;
    if (prevKey && nextKey && prevKey !== nextKey)
        return Number(nextKey) - Number(prevKey);
    if (prevKey && !nextKey)
        return -1;
    if (!prevKey && nextKey)
        return 1;
    return compareDate(pageA.routeMeta.date, pageB.routeMeta.date);
};
const getBlogOptions = (options) => ({
    ...defaultOptions,
    ...(typeof options === "object" ? options : {}),
});
exports.getBlogOptions = getBlogOptions;
const resolveBlogOptions = (options) => {
    if (!options)
        return false;
    const blogOptions = {
        ...defaultOptions,
        ...(typeof options === "object" ? options : {}),
    };
    return {
        metaScope: "",
        filter: blogOptions.filter ||
            (({ frontmatter, filePathRelative, routeMeta }) => Boolean(filePathRelative) &&
                frontmatter.home !== true &&
                routeMeta.type !== "page"),
        category: [
            {
                key: "category",
                getter: ({ routeMeta }) => routeMeta.category || [],
                sorter,
                path: blogOptions.category,
                layout: "Blog",
                itemPath: blogOptions.categoryItem,
                itemLayout: "Blog",
            },
            {
                key: "tag",
                getter: ({ routeMeta }) => routeMeta.tag || [],
                sorter,
                path: blogOptions.tag,
                layout: "Blog",
                itemPath: blogOptions.tagItem,
                itemLayout: "Blog",
            },
        ],
        type: [
            {
                key: "article",
                sorter,
                filter: () => true,
                path: blogOptions.article,
                layout: "Blog",
            },
            {
                key: "encrypted",
                sorter,
                filter: ({ routeMeta }) => Boolean(routeMeta.isEncrypted),
                path: blogOptions.encrypted,
                layout: "Blog",
            },
            {
                key: "slide",
                sorter,
                filter: ({ routeMeta }) => routeMeta.type === "slide",
                path: blogOptions.slides,
                layout: "Blog",
            },
            {
                key: "star",
                sorter,
                filter: ({ frontmatter }) => Boolean(frontmatter.star),
                path: blogOptions.star,
                layout: "Blog",
            },
            {
                key: "timeline",
                sorter: (pageA, pageB) => compareDate(pageA.routeMeta.date, pageB.routeMeta.date),
                filter: ({ frontmatter, routeMeta }) => "date" in routeMeta && frontmatter.timeline !== false,
                path: blogOptions.timeline,
                layout: "Blog",
            },
        ],
    };
};
exports.resolveBlogOptions = resolveBlogOptions;
